php/Blockonomics.php

```diff
@@ -401,50 +401,58 @@ public function is_lite_mode_active(){
     }
 
     // Adds the header to the blockonomics page
-    public function load_blockonomics_header($template_name){
+    public function load_blockonomics_header($template_name, $additional_script=NULL){
         add_action('wp_enqueue_scripts', 'bnomics_enqueue_stylesheets' );
         // Don't load javascript files if no js is active
         if (!$this->is_nojs_template($template_name)) {
             add_action('wp_enqueue_scripts', 'bnomics_enqueue_scripts' );
+            
+            if (isset($additional_script)) {
+                add_action('wp_enqueue_scripts', function () use ($additional_script) {
+                    wp_add_inline_script('bnomics-checkout', $additional_script, 'before');
+                });
+            }
         }
         // Lite mode will render without wordpress theme headers
         if($this->is_lite_mode_active()){
         ?>
-          <link rel="stylesheet" type="text/css" href="<?php echo plugins_url('css/order.css', dirname(__FILE__));?>">
+            <link rel="stylesheet" type="text/css" href="<?php echo plugins_url('css/order.css', dirname(__FILE__));?>">
+            <script src="<?php echo plugins_url('js/vendors/reconnecting-websocket.min.js', dirname(__FILE__));?>"></script>
+            <script src="<?php echo plugins_url('js/vendors/qrious.min.js', dirname(__FILE__));?>"></script>
+            <script><?php echo $additional_script; ?></script>
+            <script src="<?php echo plugins_url('js/checkout.js', dirname(__FILE__));?>"></script>
         <?php
-        }else{
-          get_header();
+        } else {
+            get_header();
         }
     }
 
     // Adds the footer to the blockonomics page
     public function load_blockonomics_footer($template_name){
         // Lite mode will render without wordpress theme footers
-        if($this->is_lite_mode_active()){
-            // Only load the lite mode javascript if nojs is not active
-            if (!$this->is_nojs_template($template_name)) {
```

Comment:
We have unnecessarily moved the lite mode JS from [the bottom footer of the page](https://github.com/blockonomics/woocommerce-plugin/pull/276/files#diff-a3307e680eb76b9d93556c9ae2e5dc134b3526246480730fb8ddf49186da29ebL427) to [the top header of the page](https://github.com/blockonomics/woocommerce-plugin/pull/276/files#diff-a3307e680eb76b9d93556c9ae2e5dc134b3526246480730fb8ddf49186da29ebR420-R423). Please revert this change.

Reply:
> Moving it to footer will require the additionalScript passed to Footer as well. Why not move all JS to footer?

`wp_add_inline_script('bnomics-checkout', $additional_script, 'before');` and `<script src="<?php echo plugins_url('js/checkout.js', dirname(__FILE__));?>"></script>` should not currently be working together and will require inline usage to fix. Yes we can move loading all JS to the footer if this will be better.

```
<script>
const checkout_data = {
  crypto: '<?php echo $crypto['code']; ?>',
  crypto_amount: <?php echo $order['satoshi']; ?>,
  ...
};
</script>
<script src="checkout.js" type="text/javascript"></script>
```

Reply:
@DarrenWestwood wp_enqueue_scripts action must be called before calling get_header() in WP. Moving the code to load_blockonomics_footer will require $additional_script to be passed as well to the footer. 

Loading of script in footer is controlled by $in_footer param in [wp_enqueue_script](https://developer.wordpress.org/reference/functions/wp_enqueue_script/) method. 

I've added the [commit for reference](https://github.com/blockonomics/woocommerce-plugin/pull/276/commits/68078d7ab89b8a2146b9e4209211b8641703095d) on how it'll be done.

Reply:
Loading scripts in the head and using defer is considered best for performance as the script is loaded asynchronously and only executed after HTML parsing: https://flaviocopes.com/javascript-async-defer/#with-defer-in-the-head

Using defer in the plugin was initially introduced to remove the "Eliminate render-blocking resources" warnings from lighthouse.
While loading the script in the footer is also suitable as it does not throw this warning, the performance may be slightly affected and will require further testing. It will be simpler to use defer as this is existing method and most efficient.

Lets move the lite mode JS back into the header and add the defer attribute to keep the lite mode consistent with the standard checkout. We can then only pass additional_script to the header again as well.

Reply:
Okay